### DISCUSSION TITLE
No wagmi config found for web3modal versions 2.2.0 up to 2.3.7

### DISCUSSION THREAD

**Discussion by aalaoui12** [WalletConnect/walletconnect-monorepo#2448] [Developer Support]
I'm using wagmi 0.12.7 (since my project necessitates it as it uses the Lens Protocol API), and I know that web3modal 2.2.0 supports wagmi 0.12.x.

However, I keep getting an error saying that no wagmi config is found when I clearly have a wagmi config tag in _app.tsx. The error I am shown is attached below. Please let me know what I could possibly do - I tried deleting node_modules and package-lock.json and reinstalling, but that does not work. I also tried using web3modal version 2.3.7 instead of 2.2.0, and this issue persists.

<img width="943" alt="Screenshot 2023-05-22 at 10 21 48 PM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/25350096/66ac85ce-7833-4693-a7a5-ab884e31123f">

**Comment by glitch-txs** [comment]
Hi, can you share your deps and code in the app file?

**Reply by aalaoui12** [reply]
Here is the _app.tsx file:
```
import {
    EthereumClient,
    w3mConnectors,
    w3mProvider,
} from "@web3modal/ethereum";
import type { AppProps } from "next/app";
import { configureChains, createClient, WagmiConfig } from "wagmi";
import {
    polygon
} from "wagmi/chains";
import { publicProvider } from "wagmi/providers/public";
import "@/styles/globals.css";
import "tailwindcss/tailwind.css";
import { ModalProvider } from "@/context/modal-context";
import React from "react";
import { Web3Modal } from "@web3modal/react";

if (!process.env.NEXT_PUBLIC_PROJECT_ID) {
    throw new Error("You need to provide NEXT_PUBLIC_PROJECT_ID env variable");
}

export const projectId = process.env.NEXT_PUBLIC_PROJECT_ID;

// 2. Configure wagmi client
const chains = [polygon];
const { provider, webSocketProvider } = configureChains(chains, [
    w3mProvider({ projectId }),
]);

const wagmiClient = createClient({
    autoConnect: true,
    connectors: w3mConnectors({
        version: 1,
        chains,
        projectId,
    }),
    provider,
    webSocketProvider,
});

// 3. Configure modal ethereum client
const ethereumClient = new EthereumClient(wagmiClient, chains);

// 4. Wrap your app with WagmiProvider and add <Web3Modal /> component
function MyApp({ Component, pageProps }: AppProps) {
    return (
        <>
        <WagmiConfig client={wagmiClient}>
            <ModalProvider>
                <Component {...pageProps} />
            </ModalProvider>
        </WagmiConfig>
        <Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
        </>
    );
}

export default MyApp;
```

**Reply by finessevanes** [reply]
Hey, check out the [docs](https://docs.walletconnect.com/2.0/web3modal/react/installation#implementation). Your implementation doesn't look correct. This is how `wagmiClient` is set: 
```javascript
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({ projectId, version: 1, chains }),
  publicClient
})
```

**Reply by aalaoui12** [reply]
Yep, but I stated that I'm using wagmi 0.12.7 out of necessity

**Comment by glitch-txs** [comment]
hey @aalaoui12 I just ran commit 3f486930a4bf341c4e4d1de2548dee0982097d85 from [the web3modal-examples repo](https://github.com/WalletConnect/web3modal-examples) (the nextjs one) which uses version 2.2.1 with wagmi 0.12.2 and works fine, maybe try that version or taking a look at the commit there :)
